--- thrift/gen-py/data_manager/data_manager.py	2016-05-11 10:00:29.000000000 +0100
+++ thrift/gen-py/data_manager/data_manager.py-lsd	2016-05-11 10:02:28.000000000 +0100
@@ -137,6 +137,7 @@
 
   def recv_get_notxn(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -172,6 +173,7 @@
 
   def recv_get(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -203,6 +205,7 @@
 
   def recv_multiget_notxn(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -238,6 +241,7 @@
 
   def recv_multiget(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -269,6 +273,7 @@
 
   def recv_tpc_prepare(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -302,6 +307,7 @@
 
   def recv_tpc_prepare2(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -335,6 +341,7 @@
 
   def recv_tpc_commit(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -364,6 +371,7 @@
 
   def recv_tpc_abort(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -393,6 +401,7 @@
 
   def recv_opc_commit(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -428,6 +437,7 @@
 
   def recv_opc_assert(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -459,6 +469,7 @@
 
   def recv_store_to_disk(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
@@ -490,6 +501,7 @@
 
   def recv_load_from_disk(self):
     iprot = self._iprot
+    iprot.trans.lsd_getreply()
     (fname, mtype, rseqid) = iprot.readMessageBegin()
     if mtype == TMessageType.EXCEPTION:
       x = TApplicationException()
